contracts/utils/structs/Heap.sol

```diff
@@ -191,9 +191,18 @@ library Heap {
         uint256 value,
         function(uint256, uint256) view returns (bool) comp
     ) private {
-        uint256 lIndex = 2 * index + 1; // this could overflow uint256
-        uint256 rIndex = 2 * index + 2; // this could overflow uint256
-
+        // Check if there is a risk of overflow when commputing the indices of the child nodes. If that is the case,
+        // there cannot be child nodes in the tree, so sifting is done.
+        if (index >= type(uint256).max / 2) return;
```

Comment:
What was the rationale for introducing this change? In the past implementation it would've taken really long to overflow

Reply:
Indeed, this change avoids an overflow in cases that will, in practice, never happen.

For an overflow to happen, the heap would have to include 2**255 items, which will, in practice never happen.
Unit testing will never notice the different. Formal verification will.

Would you remove that check with some documentation? Without documentation